hoomd/mpcd/CellList.cc

```diff
@@ -509,6 +526,22 @@ void mpcd::CellList::buildCellList()
             h_embed_cell_ids->data[cur_p - N_mpcd] = bin_idx;
             }
 
+        // compute the contribution of the particle to cell velocity
+        double4 cell_vel = h_cell_vel.data[bin_idx];
+        cell_vel.x += mass_i * vel_i.x;
+        cell_vel.y += mass_i * vel_i.y;
+        cell_vel.z += mass_i * vel_i.z;
+        cell_vel.w += mass_i;
+        h_cell_vel.data[bin_idx] = make_double4(cell_vel.x, cell_vel.y, cell_vel.z, cell_vel.w);
```

Comment:
I think this will let you get a reference to the data so you can just modify each of its element:
```suggestion
        double4& cell_vel = h_cell_vel.data[bin_idx];
        cell_vel.x += mass_i * vel_i.x;
        cell_vel.y += mass_i * vel_i.y;
        cell_vel.z += mass_i * vel_i.z;
        cell_vel.w += mass_i;
```